### PR TITLE
Feat(searchbox-component): adds support for type:search suggestions for mongo

### PR DESCRIPTION
**Pull request by mohdashraf010897**
**PR Type** : `Feature + Fixes`

**Description**
- The PR integrates support for the mongo backend, the type of query for Searchbox component now switches to `search` from `suggestion` for the mongo backend, and the response is handled accordingly.
- Tested examples and other fixes.
- Added workspaces and fixed examples directory for mongo examples.
- With the Range examples, show the `accommodates` value in the results.
---
- [Demo: react-searchbox](https://www.loom.com/share/86398889763846f4a0481fadc5dfc699)
- [Demo: vue-searchbox](https://www.loom.com/share/ea5b59545f2a437187613d7515bf326b)
---
[Related Notion Card](https://www.notion.so/appbase/Searchbox-ReactiveSearch-MongoDB-support-4c6d7b189f7d46308fb0d0d9f8e17ce4)

### CODE REVIEW

File: packages/vue-searchbox/examples/with-vue-table/src/App.vue [hunk 1]
```diff
@@ -82,7 +83,10 @@
                   {{ results.time }}ms
                 </p>
                 <div v-if="!loading && !error">
-                  <v-client-table :data="results.data" :columns="columns" :options="options"></v-client-table>
+                  <v-client-table
```
Comment (bietkul) [inline packages/vue-searchbox/examples/with-vue-table/src/App.vue:86]: @mohdashraf010897 Is there any reason to remove `options` property?

File: packages/react-searchbox/examples/with-controlled-props/src/index.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ ReactDOM.render(
     url="https://appbase-demo-ansible-abxiydt-arc.searchbase.io"
     appbaseConfig={{
       recordAnalytics: true,
-      enableQueryRules: true,
+      enableQueryRules: false,
```
Comment (bietkul) [inline packages/react-searchbox/examples/with-controlled-props/src/index.js:13]: Any reason to change it?

**Review by bietkul — APPROVED** [review]

File: packages/vue-searchbox/examples/with-vue-table/src/App.vue (hunk 1 above)
Comment (mohdashraf010897) [reply packages/vue-searchbox/examples/with-vue-table/src/App.vue:86]: yes, the options weren't defined.

File: packages/react-searchbox/examples/with-controlled-props/src/index.js (hunk 2 above)
Comment (mohdashraf010897) [reply packages/react-searchbox/examples/with-controlled-props/src/index.js:13]: to switch off getting promoted results back from the backend

**Comment by mohdashraf010897** [comment]
For commits after https://github.com/appbaseio/searchbox/pull/144/commits/89089b1e7f2c1fa41b38c1f5fe8d9af8f2a0622c until https://github.com/appbaseio/searchbox/pull/144/commits/53166f3623ae15738cae97c5b0d4d1cc1c6e56ea
the demo is attached below 👇 
React-searchbox 👉  https://www.loom.com/share/c20ef4fd9a4a495ebf1e236543e88c27
Vue-searchbox 👉  https://www.loom.com/share/33878e42443e44b8951789de2dbc6eaa

**Comment by mohdashraf010897** [comment]
@bietkul @siddharthlatest 
- [x]  **Fix the clear value with keyword (lib)**
    - **Fixed** 👇🏻
        
        [https://github.com/appbaseio/searchbox/pull/144/commits/4a6338b2b9a0eaad79988fc49cb49ff1600e4ca4](https://github.com/appbaseio/searchbox/pull/144/commits/4a6338b2b9a0eaad79988fc49cb49ff1600e4ca4)
        
    - Demo:
        
        [Demo: Fix the clear value with keyword (lib)](https://www.loom.com/share/3bdfcbb8a17b4ac2b00f2a551ac5ed0b)

**Review by bietkul — APPROVED** [review]

**Pull request merged**